Sources/XcbeautifyLib/String+TeamCityEscaping.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+extension String {
+    func teamCityEscaped() -> String {
+        // According to the documentation: https://www.jetbrains.com/help/teamcity/service-messages.html#Escaped+Values
+        return self
+            .replacingOccurrences(of: "|", with: "||")
+            .replacingOccurrences(of: "'", with: "|'")
+            .replacingOccurrences(of: "\n", with: "|n")
+            .replacingOccurrences(of: "\r", with: "|r")
+            .replacingOccurrences(of: "\\u{", with: "|u{") // Assuming the unicode format in Swift is \\u{NNNN}
+            .replacingOccurrences(of: "[", with: "|[")
+            .replacingOccurrences(of: "]", with: "|]")
+    }
+}
```

Comment:
Consider moving this into the `TeamCityRenderer` file as a `private` extension. I think it might help (for discoverability and understanding) to keep its scope limited to the type where it's needed.